packages/office-ui-fabric-react/src/components/Announced/examples/Announced.QuickActions.Example.tsx

```diff
@@ -53,177 +52,162 @@ export interface IAnnouncedQuickActionsExampleItem {
+        filesize: Math.floor(Math.random() * 30).toString() + ' MB',
+      });
+    }
+    return itemsList;
+  });
+  const [renameDialogOpen, setRenameDialogOpen] = React.useState<boolean>(false);
+
+  const [dialogContent, setDialogContent] = React.useState<JSX.Element | undefined>(undefined);
+  const [announced, setAnnounced] = React.useState<JSX.Element | undefined>(undefined);
+  const [previousAnnouncedValue, setPreviousAnnouncedValue] = React.useState<JSX.Element | undefined>(undefined);
+
+  const { setTimeout } = useSetTimeout();
+
+  const onRenderRow = useConstCallback(
+    (props: IDetailsRowProps): JSX.Element => {
+      return <DetailsRow {...props} />;
+    },
+  );
+
+  const deleteItem = useConstCallback((index: number): void => {
+    setItems(items.filter((item, i) => i !== index));
+    setAnnounced(<Announced message="Item deleted" aria-live="assertive" />);
+    setPreviousAnnouncedValue(announced);
+    return;
+  });
 
-  constructor(props: {}) {
-    super(props);
+  const renameItem = useConstCallback((item: IAnnouncedQuickActionsExampleItem, index: number): void => {
+    setRenameDialogOpen(true);
+    setDialogContent(
+      <>
+        <TextField componentRef={textField} label="Rename" defaultValue={item.name} />
+        <DialogFooter>
+          <PrimaryButton
+            // eslint-disable-next-line react/jsx-no-bind
+            onClick={() => updateItemName(index)}
+            text="Save"
+          />
+        </DialogFooter>
+      </>,
+    );
+    return;
+  });
+
+  const onRenderItemColumn = React.useCallback(
+    (item: IAnnouncedQuickActionsExampleItem, index: number, column: IColumn) => {
+      const fieldContent = item[column.fieldName as keyof IAnnouncedQuickActionsExampleItem];
+      if (column.key === 'name') {
+        return (
+          <div>
+            {fieldContent}
+            <IconButton
+              menuIconProps={{ iconName: 'MoreVertical' }}
+              role="button"
+              aria-haspopup
+              aria-label="Show actions"
+              styles={iconButtonStyles}
+              menuProps={{
+                items: [
+                  {
+                    key: 'delete',
+                    text: 'Delete',
+                    onClick: () => deleteItem(index),
+                  },
+                  {
+                    key: 'rename',
+                    text: 'Rename',
+                    onClick: () => renameItem(item, index),
+                  },
+                ],
+              }}
+            />
+          </div>
+        );
+      } else {
+        return <span>{fieldContent}</span>;
+      }
+    },
+    [deleteItem, renameItem],
+  );
+
+  const updateItemName = useConstCallback((index: number) => {
+    if (textField && textField.current) {
+      const renamedItems = items;
+      renamedItems[index].name = textField.current.value || renamedItems[index].name;
+      setItems([...renamedItems]);
```

Comment:
You should copy the items before mutating. Also don't mutate the original item in place.
```suggestion
      const renamedItems = [...items];
      renamedItems[index] = { ...renamedItems[index], name: textField.current.value || renamedItems[index].name };
      setItems(renamedItems);
```